source-code/app/src/main/java/org/buildmlearn/toolkit/activity/TemplateEditor.java

```diff
@@ -776,16 +784,34 @@ private String saveDraft() {
      * @brief Start the simulator activity
      * <p/>
      * Start the simulator with the fragment returned by the selected template. Simulator is started as a new activity.
+     * String message contains file response which will be filepath if successfully saved and otherwise error message.
      */
     private void startSimulator() {
-        String filePath = saveProject();
-        if (filePath == null || filePath.equals("")) {
+        String message = saveProject();
+        if (message == null || message.equals("")) {
             Toast.makeText(this, "Build unsuccessful", Toast.LENGTH_SHORT).show();
             return;
         }
+        else if(message.equals("File already exists"))
```

Comment:
Null pointer Exception would never come as there is "if" statement above checking it. I am changing it however.